docs/getting-started/quickstart.md

```diff
@@ -1 +1,200 @@
+     -X GET \
+     -H "X-NUCLIADB-ROLES: READER"
+```
+
+#### Semantic search
+
+Text search it's fine, but nothing so special. He are a semantic
+search engine on unstructured data, so let's demonstrate it.
+
+To perform a semantic search, we need two things: to extract and save
+vectors from the stored knowledge and to pass a similar or exact
+vector to the search endpoint.
+
+As we have inserted manually (using a Python script) the vectors, we
+can search semantically using a POST request and passing an exact or a
+similar vector:
+
+```bash
+cd nucliadb_client/examples
+VECTOR=$(python simple.py --host=localhost --http=8080 --grpc=8060 --train=8040 --kb simple --print-random-vector) \
+&& curl 'http://localhost:8080/api/v1/kb/<YOUR-KB-UUID>/search' \
+     -X POST \
+     -H "X-NUCLIADB-ROLES: READER" \
+     --data-raw "{
+  \"query\": \"whatever\",
+  \"vector\": $(python simple.py --host=localhost --http=8080 --grpc=8060 --train=8040 --kb simple --print-random-vector)
+}"
+```
+
+
+## Boundless NucliaDB with Nuclia
+
+Although it's not hard to insert your text and extracted vectors from the
+NucliaDB client, achieving a good information extraction from a ML
+pipeline and process all your documents it's definitely tough.
+
+That's why we can join strenghts with Nuclia and make this task
+astonishingly easy.
+
+### Get a NucliaDB token to connect to Nuclia Understanding API™
+ 
+First of all, you need a _Nuclia Understanding API™_ key (NUA key from
+now) that you can get following these tutorials:
+
+- [Create your free Nuclia account](https://docs.nuclia.dev/docs/quick-start/create)
+- [Get a NUA key](https://docs.nuclia.dev/docs/understanding/intro#get-a-nua-key)
+
+### Start NucliaDB
+
+Once you have your NUA key, we will start NucliaDB with it
+
+```bash
+docker run -it \
+       -e LOG=INFO \
+       -e NUA_API_KEY=<YOUR-NUA-API-KEY> \
+       -p 8080:8080 \
+       -p 8060:8060 \
+       -p 8040:8040 \
+       -v nucliadb-standalone:/data \
+       nuclia/nucliadb:latest
+```
+
+### Create a Knowledge Box container
+
+```bash
+curl 'http://localhost:8080/api/v1/kbs' \
+  -X POST \
+  -H "X-NUCLIADB-ROLES: MANAGER" \
+  -H "Content-Type: application/json" \
+  --data-raw '{
+    "slug": "mykb",
+    "title": "My KB"
+  }'
+```
+
+This call will return you a Knowledge Box UUID nedded for further API calls.
+
+### Upload a file
+
+After starting NucliaDB and creating a Knowledge Box you can upload a file:
+
+```bash
+curl http://localhost:8080/api/v1/kb/<your-knowledge-box-id>/upload \
+  -X POST \
+  -H "X-NUCLIADB-ROLES: WRITER" \
+  -H "X-FILENAME: `echo -n "myfile" | base64`"
+  -T /path/to/file
+```
+
+The file will be sent to process on Nuclia. When it finishes
+processing, your local NucliaDB will receive a message with the
+extracted information. 
+
+### Search
+
+Once we finish uploading files and getting the processed results, we can do searches on
+our data. Even we can do the same exact searches as before, having now
```

Comment:
```suggestion
our data. Ee can even do the same exact searches as before, but having now
```

Reply:
Ee?

Reply:
sorry: "We can even..." is what I meant

Reply:
also, I think it's more proper: "... as before, but now having..."